odl/solvers/advanced/forward_backward.py

```diff
@@ -0,0 +1,165 @@
+        If omitted, the simpler problem will be considered, which corresponds
+        to the convex conjugate functionals of the l_i:s being zero functionals
+        and hence the gradient being the zero-operator.
+
+    Notes
+    -----
+    Strictly, we have the following conditions on the functions involved: f and
+    g_i are proper, convex and lower semicontinuous, and h is convex and
+    differentialbe with :math:`\\eta`-Lipschitz continuous gradient.
+
+    For the optional input l_i we need it to be proper, convex, lower
+    semicontinuous, and :math:`\\nu_i^{-1}`-strongly convex. The fact that l_i
+    is :math:`\\nu_i^{-1}`-strongly convex implies that the convex conjugate
+    functional of l_i is differentiable with :math:`\\nu_i`-Lipschitz
+    continuous gradient.
+
+    To guarantee convergence, the parameters ``tau``, ``sigma`` and
+    ``L`` need to satisfy
+
+    .. math::
+
+       2 * \min \{ \\frac{1}{\\tau}, \\frac{1}{\sigma_1}, \\ldots,
+       \\frac{1}{\sigma_m} \} * \min\{ \\eta, \\nu_1, \\ldots, \\nu_m  \} *
+       \\sqrt{1 - \\tau \\sum_{i=1}^n \\sigma_i ||L_i||^2} > 1
+
+    where, if the simpler problem is considered, all :math:`\\nu_i` can be
+    considered to be :math:`\\infty`.
+
+    For references on the Forward-Backward algorithm, see [BC2015]_.
+
+    For more on convex analysis including convex conjugates and
+    resolvent operators see [Roc1970]_.
+
+    For more on proximal operators and algorithms see [PB2014]_.
+    """
+
+    # Problem size
+    m = len(L)
+
+    # Validate input
+    if not all(isinstance(op, Operator) for op in L):
+        raise ValueError('`L` not a sequence of operators')
+    if not all(op.is_linear for op in L):
+        raise ValueError('not all operators in `L` are linear')
+    if not all(x in op.domain for op in L):
+        raise ValueError('`x` not in the domain of all operators')
+    if len(sigma) != m:
+        raise ValueError('len(sigma) != len(L)')
+    if len(prox_cc_g) != m:
+        raise ValueError('len(prox_cc_g) != len(L)')
+
+    # Get parameters from kwargs
+    grad_cc_l = kwargs.pop('prox_cc_l', None)
+    if grad_cc_l is not None and len(grad_cc_l) != m:
+        raise ValueError('`grad_cc_l` not same length as `L`')
+
+    # Check for unused parameters
+    if kwargs:
+        raise TypeError('unexpected keyword argument: {}'.format(kwargs))
+
+    # Pre-allocate values
+    v = [Li.range.zero() for Li in L]
+    y = x.space.zero()
+
+    # Iteratively solve
+    for k in range(niter):
+        x_old = x
+
+        tmp_1 = grad_h(x) + sum(Li.adjoint(vi) for Li, vi in zip(L, v))
+        prox_f(tau)(x - tau * tmp_1, out=x)
+        y.lincomb(2.0, x, -1, x_old)
+
+        for i in range(m):
+            # In this case gradients were given
+            if grad_cc_l is not None:
+                tmp_2 = sigma[i] * (L[i](y) + grad_cc_l[i](v[i]))
+
+            # In this case there were not. "Applying" zero-operator
```

Comment:
Your comment is probably more clear, but mathematically I think you can see it as `grad_cc_l[i]` being the zero operator (however, instead of creating a list of zero operators and using the same line `tmp_2 = sigma[i] * (L[i](y) + grad_cc_l[i](v[i]))` in all cases, I do this if-else check).